### PR TITLE
Follow Up Doc Update to DOCS-3892 

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do?

Adding the Managed Locations partial to API test, multistep API test, and browser test doc pages.

### Motivation

Follow up on DOCS-3892 

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Review by maycmlee — APPROVED** [review]
LGTM!

File: content/en/synthetics/browser_tests/_index.md [hunk 1]
```diff
@@ -93,7 +93,11 @@ Define the configuration of your browser test.
    - For a large laptop device, the dimensions are 1440 pixels x 1100 pixels. 
    - For a tablet device, the dimensions are 768 pixels x 1020 pixels.
    - For a small mobile device, the dimensions are 320 pixels x 550 pixels.  
-6. Select **managed and private locations**: Set locations around the world that are managed by Datadog to run your test from. You can also create [private locations][1] to run your browser test from custom locations or inside private networks. For a full list of locations, navigate to the [UI][2] or use the [API][3]. {{< site-region region="gov" >}}**Note**: The West US (AWS GovCloud) location is supported on the Datadog for Government site.{{< /site-region >}}
+6. Select **managed and private locations**: Set locations around the world that are managed by Datadog to run your test from. 
+
+   {{% managed-locations %}} 
+  
+   You can also create [private locations][1] to run your browser test from custom locations or inside private networks. For a full list of locations, navigate to the [UI][2] or use the [API][3].
```
[the remark this answers is not in the document]
Comment (HugoPct) [reply content/en/synthetics/browser_tests/_index.md:100]: I guess if we provide the list, it is not needed to have this sentence anymore

File: content/en/synthetics/multistep.md [hunk 2]
```diff
@@ -50,6 +50,8 @@ Multistep API tests can run from Datadog [managed locations][5] and [private loc
 
 Select the **Locations** for your Multistep API test. Multistep API tests can run from both [managed][5] and [private locations][6] depending on your preference for running the test from outside or inside your network.
 
+{{% managed-locations %}} 
+
```
Comment (HugoPct) [inline content/en/synthetics/multistep.md:54]: I prefer having the {{}} at the end of the paragraph as you wrote in this page  after the mention of PL (multistep).
What do you think?

File: content/en/synthetics/api_tests/_index.md [hunk 3]
```diff
@@ -28,17 +28,21 @@ further_reading:
 
 ## Overview
```
Comment (HugoPct) [inline content/en/synthetics/api_tests/_index.md:29]: Feels too much. No need to update this page

**Comment by HugoPct** [comment]
In the docs we were linking Managed Locations to the API docs as it was the only way to get them.
as we are adding the locations in the docs, we can remove the link to the API.
and in the API page, link to the Locations section in HTTP tests

File: content/en/synthetics/multistep.md (hunk 2 above)
Comment (alai97) [reply content/en/synthetics/multistep.md:54]: Works for me!

File: content/en/synthetics/api_tests/dns_tests.md [hunk 4]
```diff
@@ -27,7 +27,7 @@ further_reading:
 
 DNS tests allow you to proactively monitor the resolvability and lookup times of your DNS records using any nameserver. If resolution is unexpectedly slow or a DNS server answers with unexpected A, AAAA, CNAME, TXT, or MX entries, Datadog sends you an alert with details on the failure, allowing you to quickly pinpoint the root cause of the issue and fix it.
 
-DNS tests can run from both [managed][1] and [private locations][2] depending on your preference for running the test from outside or inside your network. DNS tests can run on a schedule, on-demand, or directly within your [CI/CD pipelines][3].
+DNS tests can run from both [managed][1] and [private locations][1] depending on your preference for running the test from outside or inside your network. DNS tests can run on a schedule, on-demand, or directly within your [CI/CD pipelines][2].
```
Comment (HugoPct) [inline content/en/synthetics/api_tests/dns_tests.md:30]:
```suggestion
DNS tests can run from both [managed](#select-locations) and [private locations][1] depending on your preference for running the test from outside or inside your network. DNS tests can run on a schedule, on-demand, or directly within your [CI/CD pipelines][2].
```

File: content/en/synthetics/browser_tests/_index.md [hunk 5]
```diff
@@ -93,12 +93,11 @@ Define the configuration of your browser test.
    - For a large laptop device, the dimensions are 1440 pixels x 1100 pixels. 
    - For a tablet device, the dimensions are 768 pixels x 1020 pixels.
    - For a small mobile device, the dimensions are 320 pixels x 550 pixels.  
-6. Select **managed and private locations**: Set locations around the world that are managed by Datadog to run your test from. 
+6. Select **managed and private locations**: Set locations around the world that are managed by Datadog or create [private locations][1] to run your browser test from custom locations or inside private networks.
```
Comment (HugoPct) [inline content/en/synthetics/browser_tests/_index.md:96]:
```suggestion
6. Select **managed and private locations**: Select locations around the world that are managed by Datadog or create [private locations][1] to run your browser test from custom locations or inside private networks. 
```

**Review by HugoPct — APPROVED** [review]

**Pull request merged**